### PR TITLE
Revamped Docs 

### PR DESCRIPTION
**Pull request by pluflou**

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -1,37 +1,117 @@
...
 
-There is close integration between this repository and the `Statistical Clear Sky <https://github.com/slacgismo/StatisticalClearSky>`_ repository, which provides a "clear sky model" of system output, given only measured power as an input.
 
-See `notebooks <https://github.com/slacgismo/solar-data-tools/tree/master/notebooks>`_ folder for examples.
 
-Contents
-^^^^^^^^
+This work is supported by the U.S. Department of Energy's Office of Energy Efficiency and Renewable Energy (EERE) under the Solar Energy Technologies Office Award Number 38529.
+
+.. Important:: These documentation pages are under construction. Contributions are welcome (and much appreciated!), and you can find our contribution guidelines `here <index_dev.md>`__.
```
Comment (pluflou) [inline docs/source/index.rst:36]: remove under construction

File: docs/source/project_details/tech_details.rst [hunk 2]
```diff
@@ -0,0 +1,6 @@
+*****************
+Technical Details
+*****************
+
+The `PVInsight Technical Report <https://www.osti.gov/biblio/1897181>`_ is an in-depth report that includes details of the Solar Data Tools project,
+including detailed descriptions of the tools provided by the project.
```
Comment (pluflou) [inline docs/source/project_details/tech_details.rst:6]: add paper here

File: pyproject.toml [hunk 3]
```diff
@@ -53,15 +53,17 @@ classifiers = [
 ]
 dynamic = ["version"]
 [tool.setuptools_scm]
+write_to = "solardatatools/_version.py"
```
Comment (pluflou) [inline pyproject.toml:56]: I think I need to remove this

File: pyproject.toml [hunk 4]
```diff
@@ -53,15 +53,17 @@ classifiers = [
...
 
 [tool.setuptools.packages.find]
 include = ["solardatatools*",  "pvsystemprofiler*", "statistical_clear_sky*"]
 
 [project.optional-dependencies]
 docs = [
     "sphinx",
-    "sphinx-changelog>=1.2.0",
     "sphinx_design",
+    "nbsphinx",
```
Comment (pluflou) [inline pyproject.toml:65]: check all deps

File: requirements.txt [hunk 5]
```diff
@@ -16,3 +16,8 @@ clarabel
 qss
 tqdm
 spcqe
+pydata-sphinx-theme
+sphinx-design
+nbsphinx
+pandoc
```
Comment (pluflou) [inline requirements.txt:22]: check

File: solardatatools/__init__.py [hunk 6]
```diff
@@ -1,3 +1,4 @@
+from ._version import version as __version__
```
Comment (pluflou) [inline solardatatools/__init__.py:1]: remove

File: solardatatools/_version.py
Comment (pluflou) [inline solardatatools/_version.py:1]: remove?

File: solardatatools/data_handler.py [hunk 7]
```diff
@@ -58,6 +58,44 @@ def __init__(
         how=lambda x: x.mean(),
         gmt_offset=None,
     ):
+        """
+        Solar Data Tools' main class to handle and preprocess time series data.
+
+        This class can work with both data frames and raw data matrices, allowing for
+        time series conversion, aggregation, and various other preprocessing steps.
+
+        It is initialized by passing at least either the `data_frame` parameter or the
```
Comment (pluflou) [inline solardatatools/data_handler.py:67]: remove data matrix option

File: solardatatools/data_handler.py [hunk 8]
```diff
@@ -58,6 +58,44 @@ def __init__(
...
+        `raw_data_matrix` parameter.
+
+        :param data_frame: A pandas DataFrame containing the raw data. If provided,
+                           the class will process the data frame based on the other
+                           parameters.
+        :type data_frame: pandas.DataFrame, optional
+        :param raw_data_matrix: A raw data matrix where each column is expected to
```
Comment (pluflou) [inline solardatatools/data_handler.py:74]: add "deprecated"

File: solardatatools/data_handler.py [hunk 9]
```diff
@@ -58,6 +58,44 @@ def __init__(
...
+                             used to set the DataFrame's index to a DatetimeIndex.
+                             Required if the DataFrame index is not already a DatetimeIndex.
+        :type datetime_col: str, optional
+        :param convert_to_ts: If True, converts the data frame into a time series format
+                              by using the `make_time_series` function.
+        :type convert_to_ts: bool, default=False
+        :param no_future_dates: If True, filters out any future dates from the data frame
+                                based on the current datetime.
+        :type no_future_dates: bool, default=True
+        :param aggregate: A string representing a resampling rule (e.g., 'D' for day,
```
Comment (pluflou) [inline solardatatools/data_handler.py:87]: add "feature is no longer maintained"

**Review by bmeyers — APPROVED** [review]

**Pull request merged**